src/client/containers/HomeContainer/HomeContainer.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { FunctionComponent, useEffect } from "react";
+import { Loader } from "@fluentui/react-northstar";
+import { useRouter } from "next/router";
+
+import { storage } from "../../lib";
+
+export const HomeContainer: FunctionComponent = () => {
+    const { query: { channel, id }, query, replace } = useRouter();
+
+    const getUrl = (): string => {
+        const searchParams = new URLSearchParams(query as Record<string, string>).toString();
+        if (!storage.getAccessToken()) {
+            return `/login?${searchParams}`;
+        }
+        if (id) {
+            return `/configuration/update?${searchParams}`;
+        }
+        return `/configuration/create?${searchParams}`;
+    };
```

Comment:
I think it would be good to move all URL construction logic to a central place and use that module to build URLs via parameters instead of hard coding urls and search parameters. For example `urls.ts` module:
```js
{
   configurationUpdate({ channel, id }) { .. },
   configurationCreate({ channel, id }) { .. }
}
```

PS: This comment applies to all places where we build URLs (login container, configuration containers etc.).

Reply:
👌